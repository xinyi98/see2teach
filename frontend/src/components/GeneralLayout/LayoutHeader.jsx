import { PlusOutlined } from '@ant-design/icons';
import { Layout, Button } from 'antd';
import { Link } from 'react-router-dom';
import UserDropdown from './UserDropdown';

const { Header } = Layout;

const LayoutHeader = () => {
  return (
    <Header className="site-layout-background header">
      <div className="header-items">
        <Link to="/request-review">
          <Button type="primary" icon={<PlusOutlined />} style={{ marginRight: '2rem' }}>
            Request a Review
          </Button>
        </Link>
        <UserDropdown />
      </div>
    </Header>
  );
};

export default LayoutHeader;
