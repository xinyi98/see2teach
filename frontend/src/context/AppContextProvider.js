import React, { useContext } from 'react';
import { createContext, useState } from 'react';

export const AppContext = createContext();

/**
 * Use this hook to use the user context in the application
 * @returns Context containing {user, setUser}
 */
export function useAppContext() {
  return useContext(AppContext);
}

/**
 * This provider contains the user information. Wrap around APp.
 * @param {*} param0 Children that will be placed between provider
 * @returns JSX of the provider
 */
export function AppContextProvider({ children }) {
  const [user, setUser] = useState({
    id: window.sessionStorage.getItem('user_id'),
    name: window.sessionStorage.getItem('name'),
    upi: window.sessionStorage.getItem('upi'),
  });

  return <AppContext.Provider value={{ user, setUser }}>{children}</AppContext.Provider>;
}
