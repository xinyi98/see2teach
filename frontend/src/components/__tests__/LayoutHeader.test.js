import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { AppContextProvider } from '../../context/AppContextProvider';
import LayoutHeader from '../GeneralLayout/LayoutHeader';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

test('snapshot test for header', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <AppContextProvider>
          <LayoutHeader />
        </AppContextProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
