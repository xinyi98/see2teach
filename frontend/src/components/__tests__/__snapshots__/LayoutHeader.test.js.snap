// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for header 1`] = `
<header
  className="ant-layout-header site-layout-background header"
>
  <div
    className="header-items"
  >
    <a
      href="/request-review"
      onClick={[Function]}
    >
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        style={
          Object {
            "marginRight": "2rem",
          }
        }
        type="button"
      >
        <span
          aria-label="plus"
          className="anticon anticon-plus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
        <span>
          Request a Review
        </span>
      </button>
    </a>
    <span
      className="ant-avatar ant-avatar-circle ant-avatar-image"
      style={Object {}}
    >
      <img
        onError={[Function]}
        src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
      />
    </span>
    <button
      className="ant-dropdown-trigger ant-dropdown-link"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
       
      <span
        aria-label="down"
        className="anticon anticon-down"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </button>
  </div>
</header>
`;
